### PR TITLE
Remove some redunduncy code from #SetInitContainersAndStatuses

### PR DESCRIPTION
**Pull request by xingzhou**
Remove some redunduncy code from #SetInitContainersAndStatuses

**Release note**:
```
None
```

### CODE REVIEW

File: pkg/api/v1/pod/util.go [hunk 1]
```diff
@@ -51,9 +51,7 @@ func FindPort(pod *v1.Pod, svcPort *v1.ServicePort) (int, error) {
 
 // TODO: remove this function when init containers becomes a stable feature
 func SetInitContainersAndStatuses(pod *v1.Pod) error {
-	var initContainersAnnotation string
-	initContainersAnnotation = pod.Annotations[v1.PodInitContainersAnnotationKey]
-	initContainersAnnotation = pod.Annotations[v1.PodInitContainersBetaAnnotationKey]
+	initContainersAnnotation := pod.Annotations[v1.PodInitContainersBetaAnnotationKey]
```
Comment (k82cn) [inline pkg/api/v1/pod/util.go]: please refer to `Convert_v1_PodTemplateSpec_To_api_PodTemplateSpec` on how to handle this.

Comment (xingzhou) [inline pkg/api/v1/pod/util.go]: Any reason to follow the logic in `Convert_v1_PodTemplateSpec_To_api_PodTemplateSpec`? this method is called in two places, one is to convey the pod's container and status field, the other one is in the test to construct the pod object, is there any special process that need to take care of?

Comment (xingzhou) [inline pkg/api/v1/pod/util.go]: for compatibility of pods without `beta` annotation?

**Comment by k82cn** [comment]
hm... , it seems a duplicated PR with https://github.com/kubernetes/kubernetes/pull/41005 .

**Comment by xingzhou** [comment]
ok, I can close this one.

**Pull request closed without merging**